Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -851,9 +931,294 @@ struct HFTrackIndexSkimsCreator {
+          continue;
+        }
+
+        // invariant mass
+        // calculate invariant masses
+        auto arrMom = array{pVecBach, pVecV0};
+        mass2K0sP = RecoDecay::M(arrMom, array{massP, massK0s});
+        // invariant-mass cut
+        if ((d_cutCascInvMassLc >= 0.) && (std::abs(mass2K0sP - massLc) > d_cutCascInvMassLc)) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg << " rejected due to invMass cut: " << mass2K0sP << ", mass Lc " << massLc << " (cut " << d_cutCascInvMassLc << ")");
+          continue;
+        }
+
+        std::array<float, 3> posCasc = {0., 0., 0.};
+        const auto& cascVtx = fitter.getPCACandidate();
+        for (int i = 0; i < 3; i++) {
+          posCasc[i] = cascVtx[i];
+        }
+
+        // fill table row
+        rowTrackIndexCasc(bach.globalIndex(),
+                          v0.globalIndex(),
+                          1); // 1 should be the value for the Lc
+        // fill histograms
+        if (b_doValPlots) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc && isLc, LOG(INFO) << "KEPT! True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg);
+          registry.get<TH1>(HIST("hvtx2_x"))->Fill(posCasc[0]);
+          registry.get<TH1>(HIST("hvtx2_y"))->Fill(posCasc[1]);
+          registry.get<TH1>(HIST("hvtx2_z"))->Fill(posCasc[2]);
+          registry.get<TH1>(HIST("hmass2"))->Fill(mass2K0sP);
+        }
+
+      } // loop over V0s
+
+    } // loop over tracks
+  }   // process
+};
+
+struct HFTrackIndexTestMCCasc {
+  void process(aod::BigTracksMC const& tracks,
+               aod::V0Datas const& V0s,
+               aod::McParticles const& particlesMC)
+
+  {
+    int8_t sign = 0;
+    for (const auto& v0 : V0s) {
+      // selections on the V0 daughters
+      auto arrayDaughtersV0 = array{v0.posTrack_as<aod::BigTracksMC>(), v0.negTrack_as<aod::BigTracksMC>()};
+      LOG(INFO) << "Looking for K0s ";
+      RecoDecay::getMatchedMCRec(particlesMC, arrayDaughtersV0, 310, array{+kPiPlus, -kPiPlus}, true, &sign, 1); // does it matter the "acceptAntiParticle" in the K0s case? In principle, there is no anti-K0s
+    }
+    return;
+  }
+};
+
+//________________________________________________________________________________________________________________________
+
 WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
 {
-  return WorkflowSpec{
+  WorkflowSpec workflow{
     adaptAnalysisTask<SelectTracks>(cfgc, "hf-produce-sel-track"),
-    adaptAnalysisTask<HFTrackIndexSkimsCreator>(cfgc, "hf-track-index-skims-creator")};
+    adaptAnalysisTask<HFTrackIndexSkimsCreator>(cfgc, "hf-track-index-skims-creator"),
+    adaptAnalysisTask<HFTrackIndexSkimsCreatorCascades>(cfgc, "hf-track-index-skims-cascades-creator")};
```

Comment:
I think the cascade skimming should be optional. Otherwise it always runs, even when only the standard 2/3-prong skimming is wanted and one has to run `lambdakzerobuilder` to produce `V0Datas`.

Reply:
Done